graphics/managed_surface.h

```diff
@@ -303,7 +324,7 @@ class ManagedSurface {
 	 * Draw a thick line.
 	 */
 	void drawThickLine(int x0, int y0, int x1, int y1, int penX, int penY, uint32 color) {
-		_innerSurface.drawThickLine(x0, y0, x1, y1, penX, penY, color);
+		_innerSurface.drawThickLine(x0, y0, x1 + penX, y1 + penY, penX, penY, color);
```

Comment:
I think you meant to change the addDirtyRect call here rather than the draw call.